jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -904,8 +905,17 @@ def _fit_model_to_spectrum(self, add_data):
             return
         models_to_fit = self._reinitialize_with_fixed()
 
-        masked_spectrum = self._apply_subset_masks(self.dataset.selected_spectrum,
+        spec = self.dataset.selected_spectrum
+        # Needs Attention:
+        # should conversion occur before or after call to _apply_subset_masks?
+        if spec.flux.unit.to_string != self.app._get_display_unit('flux'):
+            equivalencies = u.spectral_density(self.dataset.selected_spectrum.spectral_axis)
+            spec = self.dataset.selected_spectrum.with_flux_unit(self.app._get_display_unit('flux'),
+                                                                 equivalencies=equivalencies)
```

Comment:
should this all be moved into the `DatasetSelect.selected_spectrum` logic so its reusable and cached (and then the cache would need to be invalidated for a change in flux display units)?

Reply:
I added some logic to `handle_display_unit`, and with that, this block is removed and selected_spectrum(use_display_units=True) replaces it.